src/platform/forms-system/src/js/fields/ObjectField.jsx

```diff
@@ -197,24 +197,22 @@ class ObjectField extends React.Component {
       'schemaform-block': title && !isRoot,
     });
 
+    const pageIndex = formContext?.pagePerItemIndex;
     // Fix array nested ids (one-level deep)
     const processIds = (originalIds = {}) =>
       // pagePerItemIndex is zero-based
-      typeof formContext?.pagePerItemIndex !== 'undefined' &&
-      formContext.onReviewPage
+      typeof pageIndex !== 'undefined' && formContext.onReviewPage
         ? Object.keys(originalIds).reduce(
             (ids, key) => ({
               ...ids,
               [key]: originalIds[key].$id
                 ? {
                     ...originalIds[key],
-                    $id: `${originalIds[key].$id}_${
-                      formContext.pagePerItemIndex
-                    }`,
+                    $id: `${originalIds[key].$id}_${pageIndex}`,
                   }
-                : originalIds[key],
+                : `${originalIds[key]}_${pageIndex}`,
             }),
-            {},
+            { $id: `${originalIds.$id}_${pageIndex}` },
```

Comment:
Why does the reducer need this initial object as opposed to an empty one?

Reply:
Hmm, you're right.. I thought I had a case where the root `$id` wasn't being updated, but I couldn't find it. So I'll remove it.